### PR TITLE
[REVIEW] Fix #400 DataFrame initialization silently fails for generated column

### LINKED ISSUE #400: DataFrame initialization silently fails for generated column
Initializing a DataFrame silently fails when using a generator for a column:

```
>>> from pygdf import DataFrame as gdf
>>> df = gdf({
...   'k': (i for i in range(5))
... })
>>> 
>>> print(df)
Empty DataFrame
```

System details are below.  This is the RAPIDS image pulled from DockerHub:

```
(gdf) root@8bac9b8e1c55:/rapids/cudf-alpha# git log | head -3
commit c7aed5f43fed571a3e8d7af34918a11ae242f49a
Author: Mike Wendt <mwendt@nvidia.com>
Date:   Thu Oct 11 15:07:39 2018 +0200
(gdf) root@8bac9b8e1c55:/rapids/cudf-alpha# python --version
Python 3.5.6 :: Anaconda, Inc.
(gdf) root@8bac9b8e1c55:/rapids/cudf-alpha# uname -a
Linux 8bac9b8e1c55 4.4.0-137-generic #163-Ubuntu SMP Mon Sep 24 13:14:43 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
(gdf) root@8bac9b8e1c55:/rapids/cudf-alpha# nvidia-smi 
Wed Nov 21 18:47:09 2018       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 396.44                 Driver Version: 396.44                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla P100-SXM2...  Off  | 00000000:05:00.0 Off |                    0 |
| N/A   35C    P0    43W / 300W |   2290MiB / 16280MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla P100-SXM2...  Off  | 00000000:06:00.0 Off |                    0 |
| N/A   30C    P0    33W / 300W |     10MiB / 16280MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla P100-SXM2...  Off  | 00000000:84:00.0 Off |                    0 |
| N/A   28C    P0    34W / 300W |     10MiB / 16280MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   3  Tesla P100-SXM2...  Off  | 00000000:85:00.0 Off |                    0 |
| N/A   32C    P0    34W / 300W |     10MiB / 16280MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
+-----------------------------------------------------------------------------+
```

### PR DESCRIPTION
**Pull request by trstovall**
DataFrame initialization silently failed for generated columns.  This is because generators may only be consumed once.  This fix intercepts the generator and converts to Series.

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by harrism** [comment]
Can you please give this PR a descriptive title? Plan for something that will go in the CHANGELOG.md.

**Comment by harrism** [comment]
run tests

**Comment by harrism** [comment]
@trstovall in the future I recommend you create a topic branch in your fork for any development you plan to put into a pull request. If you PR from master, then any push you do to your fork's master branch will be added to the PR. See https://help.github.com/articles/about-pull-requests/

**Comment by randerzander** [comment]
@trstovall the style failure is basically running "flake8" from the cudf/python directory. If you conda or pip install flake8 and run it locally, you'll get the same information. Fixing those errors would address the style CI failure.

**Comment by kkraus14** [comment]
run tests

**Comment by kkraus14** [comment]
Looks like a minor style issue:
```
./python/cudf/tests/test_dataframe.py:1010:1: W293 blank line contains whitespace
```
And you need to add a line to the CHANGELOG.md file in order for CI to pass 😄.

**Comment by kkraus14** [comment]
I resolved the style issues and added a changelog entry for you.

**Comment by kkraus14** [comment]
run tests

**Comment by trstovall** [comment]
The patch assumed columns to be added were iterable sequences.  Tests are failing where scalars expected to be converted to sequences.

**Comment by trstovall** [comment]
run tests

File: python/cudf/dataframe/dataframe.py [hunk 1]
```diff
@@ -565,7 +565,8 @@ def add_column(self, name, data, forceindex=False):
         if name in self._cols:
             raise NameError('duplicated column name {!r}'.format(name))
 
-        series = Series(data)
+        if isinstance(data, type(1 for i in range(1))):
+            series = Series(data)
```
Comment (harrism) [inline python/cudf/dataframe/dataframe.py]: You need an else here to initialize `series`. This is the cause of your current test failures.

Comment (harrism) [reply python/cudf/dataframe/dataframe.py]: Please run the tests locally -- running pytests locally should have tripped this before you pushed.

**Comment by randerzander** [comment]
@trstovall in the interests of burning down issues for the .5 release quickly, are you able to make the changes @harrism suggested or are you ok with someone else driving this fix to completion?

**Comment by trstovall** [comment]
I can fix and test tomorrow morning.

**Comment by trstovall** [comment]
Please rerun tests.  If tests fail, please delegate someone else to drive this forward.

**Comment by trstovall** [comment]
Regarding build error:
```
# ls -al /usr/lib/x86_64-linux-gnu/
...
lrwxrwxrwx 1 root root       17 Dec  6 21:24 libcuda.so -> libcuda.so.396.44
lrwxrwxrwx 1 root root       17 Jan 15 15:11 libcuda.so.1 -> libcuda.so.410.79
# nvidia-smi
Tue Jan 15 16:03:09 2019       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 410.79       Driver Version: 410.79       CUDA Version: 10.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla P100-SXM2...  Off  | 00000000:05:00.0 Off |                    0 |
| N/A   36C    P0    43W / 300W |   8447MiB / 16280MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla P100-SXM2...  Off  | 00000000:06:00.0 Off |                    0 |
| N/A   33C    P0    44W / 300W |   8447MiB / 16280MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla P100-SXM2...  Off  | 00000000:84:00.0 Off |                    0 |
| N/A   30C    P0    44W / 300W |   8447MiB / 16280MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   3  Tesla P100-SXM2...  Off  | 00000000:85:00.0 Off |                    0 |
| N/A   35C    P0    44W / 300W |   8447MiB / 16280MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
+-----------------------------------------------------------------------------+
```

**Comment by kkraus14** [comment]
rerun tests

**Comment by kkraus14** [comment]
rerun tests

**Comment by kkraus14** [comment]
rerun tests

**Pull request merged**